src/main/kotlin/com/ajaxproject/warehouse/service/WaybillServiceImpl.kt

```diff
@@ -25,6 +32,46 @@ class WaybillServiceImpl(
             .mapToDataDto()
     }
 
+    override fun createWaybill(createDto: WaybillCreateDto): WaybillDataDto {
+        val waybill = createDto.mapToEntity()
+        if (createDto.products == null) {
+            return waybillRepository.save(waybill).mapToDataDto()
+        }
+        val errorList: MutableList<String> = mutableListOf()
+        val products: MutableList<Pair<Product, Int>> = mutableListOf()
+        for (productDto in createDto.products) {
+            productRepository.findById(productDto.id as Int).ifPresentOrElse(
+                { products.add(it to productDto.amount as Int) },
+                { errorList.add("Product with id ${productDto.id} not found") }
+            )
+        }
+        if (errorList.isNotEmpty()) {
+            throw NotFoundException(errorList)
+        }
+        waybillRepository.save(waybill)
+        products.forEach { (product, amount) ->
+            waybillProductRepository.saveProductAndChangeAmount(waybill, product, amount)
+        }
+        return waybill.mapToDataDto()
+    }
+
+    fun WaybillProductRepository.saveProductAndChangeAmount(waybill: Waybill,
+                                                            product: Product,
+                                                            amount: Int) {
```

Comment:
```suggestion
    fun WaybillProductRepository.saveProductAndChangeAmount(
     waybill: Waybill,
     product: Product,
     amount: Int
     ) {
```

this is kotlin styoe